### ISSUE TITLE
Problems with conda env .yaml

### ISSUE THREAD

**Issue by CHoeltermann** [Chelysheva/RNA2HLA#2]
Problem: Conda environment yaml throws unsolveable conflict error when being used to create a new environment.
Conda version: 4.9.2
OS: linux (Ubuntu LTS 20.04)

After trying to install the dependencies manually into a conda env, I realized numpy 1.16.6 seems to be the problem. Please review.
```

user@thing:~ conda env create -f RNA2HLA_env.yml 
Collecting package metadata (repodata.json): done
Solving environment: / 
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed                                                                                                                                                                                                   /  
Solving environment: \ 
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed                                                                                                                                                                                                   |  

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

Package numpy conflicts for:
numpy=1.16.6
pandas=0.24.2 -> numpy[version='>=1.12.1,<2.0a0|>=1.14.6,<2.0a0']
biopython=1.76 -> numpy

Package pypy3.6 conflicts for:
python-dateutil=2.8.1 -> python -> pypy3.6[version='7.3.0.*|7.3.1.*|7.3.2.*|7.3.3.*']
numpy=1.16.6 -> python[version='>=3.6,<3.7.0a0'] -> pypy3.6[version='7.3.0.*|7.3.1.*|7.3.2.*|7.3.3.*']
pandas=0.24.2 -> numpy[version='>=1.12.1,<2.0a0'] -> pypy3.6[version='7.3.0.*|7.3.1.*|7.3.2.*|7.3.3.*|>=7.3.1|>=7.3.2|>=7.3.3']
pip=19.3.1 -> setuptools -> pypy3.6[version='7.3.0.*|7.3.1.*|7.3.2.*|7.3.3.*|>=7.3.1|>=7.3.2|>=7.3.3']
biopython=1.76 -> numpy -> pypy3.6[version='7.3.0.*|7.3.1.*|7.3.2.*|7.3.3.*|>=7.3.1|>=7.3.2|>=7.3.3']

Package python conflicts for:
python=2.7
conda-forge::r-base=3.6.3 -> glib[version='>=2.58.3,<3.0a0'] -> python[version='*|>=2.7|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=3.8,<3.9.0a0|>=2.7,<2.8.0a0']
pandas=0.24.2 -> numpy[version='>=1.12.1,<2.0a0'] -> python[version='2.7.*|3.5.*|3.6.*|>=3.5,<3.6.0a0|>=3.9,<3.10.0a0|3.4.*|>=3']
numpy=1.16.6 -> python[version='>=2.7,<2.8.0a0|>=3.9,<3.10.0a0|>=3.8,<3.9.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0']
biopython=1.76 -> numpy -> python[version='2.7.*|3.5.*|3.6.*|>=3.5,<3.6.0a0|>=3.9,<3.10.0a0|3.4.*']
pip=19.3.1 -> python[version='>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.8,<3.9.0a0|>=3.7,<3.8.0a0']
bioconda::bowtie=1.1.2 -> python[version='2.7.*|3.4.*|3.5.*']
biopython=1.76 -> python[version='>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=3.8,<3.9.0a0']
numpy=1.16.6 -> mkl_fft[version='>=1.0.6,<2.0a0'] -> python[version='2.7.*|3.5.*|3.6.*|>=3.5,<3.6.0a0|3.7.*|3.8.*']
pandas=0.24.2 -> python[version='>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=3.8,<3.9.0a0']
pip=19.3.1 -> setuptools -> python[version='2.7.*|3.5.*|3.6.*|>=3.5,<3.6.0a0|>=3.9,<3.10.0a0|3.4.*']

Package setuptools conflicts for:
python=2.7 -> pip -> setuptools
pip=19.3.1 -> setuptools

Package libgfortran5 conflicts for:
numpy=1.16.6 -> blas=[build=openblas] -> libgfortran5[version='>=9.3.0']
conda-forge::r-base=3.6.3 -> libgfortran-ng -> libgfortran5[version='8.4.0.*|9.3.0.*']
conda-forge::r-base=3.6.3 -> libgfortran5[version='>=9.3.0']

Package python_abi conflicts for:
pip=19.3.1 -> setuptools -> python_abi[version='3.6|3.6.*|3.7|3.7.*|3.9.*|3.8.*',build='*_cp38|*_cp36m|*_pypy37_pp73|*_cp37m|*_cp39|*_pypy36_pp73']
numpy=1.16.6 -> python_abi[version='3.6.*|3.7.*|3.8.*',build='*_cp38|*_cp36m|*_cp37m']
numpy=1.16.6 -> mkl-service[version='>=2,<3.0a0'] -> python_abi[version='3.6|3.9.*|3.7',build='*_pypy37_pp73|*_cp39|*_pypy36_pp73']
conda-forge::r-base=3.6.3 -> glib[version='>=2.58.3,<3.0a0'] -> python_abi[version='2.7.*|3.6|3.6.*|3.7.*|3.8.*',build='*_cp27mu|*_cp38|*_cp36m|*_cp37m|*_pypy36_pp73']
biopython=1.76 -> numpy -> python_abi[version='3.6.*|3.6|3.7.*|3.7|3.9.*|3.8.*',build='*_cp36m|*_cp38|*_cp39|*_pypy37_pp73|*_cp37m|*_pypy36_pp73']
pandas=0.24.2 -> numpy[version='>=1.12.1,<2.0a0'] -> python_abi[version='3.6.*|3.6|3.7.*|3.7|3.9.*|3.8.*',build='*_cp36m|*_cp38|*_cp39|*_pypy37_pp73|*_cp37m|*_pypy36_pp73']
python-dateutil=2.8.1 -> python -> python_abi[version='3.6|3.7',build='*_pypy37_pp73|*_pypy36_pp73']

Package openblas conflicts for:
conda-forge::r-base=3.6.3 -> libblas[version='>=3.8.0,<4.0a0'] -> openblas[version='0.3.5.*|>=0.3.6,<0.3.7.0a0']
biopython=1.76 -> numpy -> openblas[version='0.2.18.*|0.2.18|0.2.18.*|0.2.19|0.2.19.*|0.2.20|0.2.20.*|>=0.2.20,<0.2.21.0a0|>=0.3.3,<0.3.4.0a0|>=0.2.15']
numpy=1.16.6 -> blas=[build=openblas] -> openblas[version='0.3.5.*|>=0.3.6,<0.3.7.0a0']
pandas=0.24.2 -> numpy[version='>=1.12.1,<2.0a0'] -> openblas[version='0.2.19|0.2.19.*|0.2.20|0.2.20.*|>=0.2.20,<0.2.21.0a0|>=0.3.3,<0.3.4.0a0']

Package liblapack conflicts for:
numpy=1.16.6 -> liblapack[version='>=3.8.0,<4.0a0']
numpy=1.16.6 -> blas=[build=openblas] -> liblapack[version='3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.9.0|3.9.0|3.9.0|3.9.0',build='4_openblas|7_openblas|11_openblas|13_openblas|14_openblas|17_openblas|7_openblas|8_openblas|6_openblas|5_openblas|16_openblas|15_openblas|12_openblas|10_openblas|9_openblas|8_openblas|6_openblas|5_openblas|3_openblas|2_openblas|0_openblas']

Package xz conflicts for:
biopython=1.76 -> python[version='>=3.7,<3.8.0a0'] -> xz[version='5.2.*|>=5.2.3,<5.3.0a0|>=5.2.4,<5.3.0a0|>=5.2.4,<6.0a0|>=5.2.5,<6.0a0|>=5.2.5,<5.3.0a0|>=5.2.3,<6.0a0']
bioconda::bowtie=1.1.2 -> python=3.5 -> xz[version='5.0.*|5.2.*|>=5.2.3,<5.3.0a0|>=5.2.3,<6.0a0|>=5.2.4,<6.0a0']
conda-forge::r-base=3.6.3 -> xz[version='>=5.2.4,<5.3.0a0|>=5.2.5,<5.3.0a0']
pandas=0.24.2 -> python[version='>=3.6,<3.7.0a0'] -> xz[version='5.2.*|>=5.2.3,<5.3.0a0|>=5.2.3,<6.0a0|>=5.2.4,<6.0a0|>=5.2.5,<6.0a0|>=5.2.5,<5.3.0a0|>=5.2.4,<5.3.0a0']
conda-forge::r-base=3.6.3 -> libtiff[version='>=4.2.0,<5.0a0'] -> xz[version='>=5.2.4,<6.0a0|>=5.2.5,<6.0a0']
python-dateutil=2.8.1 -> python -> xz[version='5.0.*|5.2.*|>=5.2.3,<5.3.0a0|>=5.2.3,<6.0a0|>=5.2.4,<6.0a0|>=5.2.5,<6.0a0|>=5.2.5,<5.3.0a0|>=5.2.4,<5.3.0a0']
pip=19.3.1 -> python[version='>=3.8,<3.9.0a0'] -> xz[version='5.2.*|>=5.2.3,<5.3.0a0|>=5.2.3,<6.0a0|>=5.2.4,<5.3.0a0|>=5.2.4,<6.0a0|>=5.2.5,<6.0a0|>=5.2.5,<5.3.0a0']
numpy=1.16.6 -> python[version='>=3.9,<3.10.0a0'] -> xz[version='5.2.*|>=5.2.3,<5.3.0a0|>=5.2.3,<6.0a0|>=5.2.4,<5.3.0a0|>=5.2.4,<6.0a0|>=5.2.5,<5.3.0a0|>=5.2.5,<6.0a0']

Package six conflicts for:
numpy=1.16.6 -> mkl-service[version='>=2,<3.0a0'] -> six
pandas=0.24.2 -> python-dateutil[version='>=2.5.*'] -> six[version='>=1.5']
python-dateutil=2.8.1 -> six[version='>=1.5']

Package libgfortran-ng conflicts for:
numpy=1.16.6 -> blas=[build=openblas] -> libgfortran-ng
numpy=1.16.6 -> libgfortran-ng[version='>=7,<8.0a0']

Package libcblas conflicts for:
numpy=1.16.6 -> blas=[build=openblas] -> libcblas[version='3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.9.0|3.9.0|3.9.0|3.9.0',build='4_openblas|7_openblas|11_openblas|13_openblas|14_openblas|17_openblas|7_openblas|8_openblas|6_openblas|5_openblas|16_openblas|15_openblas|12_openblas|10_openblas|9_openblas|8_openblas|6_openblas|5_openblas|3_openblas|2_openblas|0_openblas']
numpy=1.16.6 -> libcblas[version='>=3.8.0,<4.0a0']

Package libgfortran conflicts for:
pandas=0.24.2 -> numpy[version='>=1.12.1,<2.0a0'] -> libgfortran[version='>=3.0']
biopython=1.76 -> numpy -> libgfortran[version='>=3.0']

Package _libgcc_mutex conflicts for:
numpy=1.16.6 -> libgcc-ng[version='>=7.3.0'] -> _libgcc_mutex[version='*|0.1',build='conda_forge|main']
python=2.7 -> libgcc-ng[version='>=7.3.0'] -> _libgcc_mutex[version='*|0.1',build='conda_forge|main']
pandas=0.24.2 -> libgcc-ng[version='>=7.3.0'] -> _libgcc_mutex[version='*|0.1',build='conda_forge|main']
conda-forge::r-base=3.6.3 -> libgcc-ng[version='>=9.3.0'] -> _libgcc_mutex[version='*|0.1',build='conda_forge|main']
biopython=1.76 -> libgcc-ng[version='>=7.3.0'] -> _libgcc_mutex[version='*|0.1',build='conda_forge|main']

Package pip conflicts for:
pip=19.3.1
numpy=1.16.6 -> python[version='>=3.9,<3.10.0a0'] -> pip
python=2.7 -> pip
biopython=1.76 -> python[version='>=3.7,<3.8.0a0'] -> pip
bioconda::bowtie=1.1.2 -> python=3.5 -> pip
pandas=0.24.2 -> python[version='>=3.6,<3.7.0a0'] -> pip

Package libblas conflicts for:
numpy=1.16.6 -> libblas[version='>=3.8.0,<4.0a0']
numpy=1.16.6 -> blas=[build=openblas] -> libblas[version='3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0.*|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.8.0|3.9.0|3.9.0|3.9.0|3.9.0|3.9.0|3.9.0|3.9.0|3.9.0|3.9.0|3.9.0|3.9.0|3.9.0|3.9.0|3.9.0|3.9.0|3.9.0|3.9.0.*|3.9.0|3.9.0|3.9.0|3.9.0',build='4_blis|4_mkl|5_blis|5_mkl|7_blis|9_blis|10_blis|12_blis|12_mkl|13_mkl|13_blis|14_blis|16_blis|20_mkl|0_blis|1_openblas|2_blis|3_blis|4_blis|4_openblas|4_mkl|5_blis|6_blis|6_mkl|7_blis|8_mkl|4_openblas|7_openblas|11_openblas|13_openblas|14_openblas|17_openblas|7_openblas|8_openblas|6_openblas|5_openblas|16_openblas|15_openblas|12_openblas|10_openblas|9_openblas|8_openblas|6_openblas|5_openblas|3_openblas|2_openblas|0_openblas|8_blis|7_mkl|5_mkl|1_blis|21_mkl|19_mkl|18_mkl|16_mkl|15_mkl|15_blis|14_mkl|11_mkl|11_blis|10_mkl|9_mkl|8_mkl|8_blis|7_mkl|6_mkl|6_blis']

Package libopenblas conflicts for:
numpy=1.16.6 -> libblas[version='>=3.8.0,<4.0a0'] -> libopenblas[version='>=0.3.10,<0.3.11.0a0|>=0.3.12,<0.3.13.0a0|>=0.3.12,<1.0a0|>=0.3.10,<1.0a0|>=0.3.9,<0.3.10.0a0|>=0.3.9,<1.0a0|>=0.3.8,<0.3.9.0a0|>=0.3.8,<1.0a0|>=0.3.7,<0.3.8.0a0|>=0.3.7,<1.0a0|>=0.3.6,<0.3.7.0a0|>=0.3.6,<1.0a0']
numpy=1.16.6 -> libopenblas[version='>=0.3.3,<1.0a0']

Package python-dateutil conflicts for:
python-dateutil=2.8.1
pandas=0.24.2 -> python-dateutil[version='>=2.5.*|>=2.6.1']

Package mkl conflicts for:
numpy=1.16.6 -> blas==1.0=mkl -> mkl[version='2019.1.*|>=2018.0.3,<2019.0a0|>=2019.1,<2021.0a0|>=2019.3,<2020.0a0|>=2019.3,<2021.0a0|>=2020.2,<2021.0a0|>=2019.4,<2020.0a0|>=2020.4,<2021.0a0|>=2019.1,<2020.0a0|>=2020.1,<2021.0a0|>=2020.0,<2021.0a0|>=2019.0,<2020.0a0']
numpy=1.16.6 -> mkl[version='>=2019.4,<2021.0a0']

Package libedit conflicts for:
conda-forge::r-base=3.6.3 -> krb5[version='>=1.17.1,<1.18.0a0'] -> libedit[version='>=3.1.20170329,<4.0a0|>=3.1.20181209,<3.2.0a0|>=3.1.20181209,<4.0a0|>=3.1.20191231,<3.2.0a0|>=3.1.20191231,<4.0a0']
python=2.7 -> sqlite[version='>=3.31.1,<4.0a0'] -> libedit[version='>=3.1.20170329,<3.2.0a0|>=3.1.20181209,<3.2.0a0|>=3.1.20191231,<3.2.0a0']

Package tzdata conflicts for:
python-dateutil=2.8.1 -> python -> tzdata
numpy=1.16.6 -> python[version='>=3.9,<3.10.0a0'] -> tzdata
Note that strict channel priority may have removed packages required for satisfiability.

```

**Comment by Chelysheva** [comment]
Thank you for reporting this issue!
The environment file has been just downloaded for GitHub rep and re-tested from scratch on 2 independent computers (ubuntu 16.04 and 18.04) and the environment has been created (both - conda 4.9.2) and activated successfully. While testing on the other computer (MacOS Mojave) - an error similar to yours has appeared. It has been solved by changing numpy=1.16.1 to numpy=1.15 in environment file. Hope, this would resolve your issue as well. I would add this option and comment to the description.

**Comment by CHoeltermann** [comment]
Thank you for the quick help! This indeed solved the issue on my machine.

All the best,
Charlotte

Irina Chelysheva ***@***.***> schrieb am Fr., 2. Apr. 2021,
17:24:
